node/metered-channel/src/bounded.rs

```diff
@@ -76,6 +73,22 @@ impl<T> Stream for MeteredReceiver<T> {
 }
 
 impl<T> MeteredReceiver<T> {
+	fn maybe_meter_tof(&mut self, maybe_value: Option<MaybeTimeOfFlight<T>>) -> Option<T> {
+		self.meter.note_received();
+		maybe_value.map(|value| {
+			match value {
+				MaybeTimeOfFlight::<T>::WithTimeOfFlight(value, tof_start) => {
+					// do not use `.elapsed()`, it may panic
```

Comment:
It uses monotonic `clock_gettime` so subsequent calls must be >= comparing to the current call even in the case of NTP influence. So probably `tof_start.elapsed()` is totally safe. However, it does not save anything...

Reply:
We can use https://github.com/jedisct1/rust-coarsetime if we want something really fast btw.

Reply:
Histogram timers also have a coarse option which might make use of this

Reply:
https://docs.rs/prometheus/latest/src/prometheus/histogram.rs.html#497-501 it's not enabled unless built with nightly.